builder error when an operator of a generic struct is used but not defined

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Linux version 5.4.157-17185-gbd27b903c738 (chrome-bot@localhost) (Chromium OS 13.0_pre433403_p20211019-r1 clang version 13.0.0 (/var/tmp/portage/sys-devel/llvm-13.0_pre433403_p20211019-r1/work/llvm-13.0_pre433403_p20211019/clang 0fe7531b75e54909a7cb7f7080634ab2053964ad)) #1 SMP PREEMPT Sat Jan 1 19:17:16 PST 2022
Processor: 4 cpus, 64bit, little endian, aarch64
CC version: cc (Debian 10.2.1-6) 10.2.1 20210110

getwd: /home/code/v/2d_walking
vmodules: /home/.vmodules
vroot: /home/v
vexe: /home/v/v
vexe mtime: 2022-01-20 00:46:54
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 35418b8.ba33082

Git version: git version 2.30.2
Git vroot status: weekly.2022.03-15-gba330829
.git/config present: true
thirdparty/tcc status: thirdparty-linux-aarch64 b2d41aba

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg error.v`
```v

struct V2d<T>{
mut:
	x T
	y T
}

fn main(){

	v1 := V2d<int>{2 2}
	v2 := V2d<int>{2 3}
	v3 := V2d<int>{4 6}
	if v1 * v2 == v3{

	}
	// assert v1 * v2 == v3
}
```

**What did you expect to see?**

graceful error as the * operator is not implemented yet


**What did you see instead?**
```
==================
/tmp/v_1000/error.422440444420909900.tmp.c:11076: warning: implicit declaration of function 'main__V2d_T_int__mult_T_int'
/tmp/v_1000/error.422440444420909900.tmp.c:11076: error: cannot convert 'int' to 'struct main__V2d_T_int'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -679,31 +679,25 @@ pub fn (mut c Checker) infix_expr(mut node ast.InfixExpr) ast.Type {
 				}
 			}
 			if left_sym.kind in [.array, .array_fixed, .map, .struct_] {
-				if left_sym.has_method(node.op.str()) {
-					if method := left_sym.find_method(node.op.str()) {
+				if left_sym.has_method_with_generic_parent(node.op.str()) {
+					if method := left_sym.find_method_with_generic_parent(node.op.str()) {
 						return_type = method.return_type
 					} else {
 						return_type = left_type
 					}
 				} else {
-					if left_sym.kind == .struct_
-						&& (left_sym.info as ast.Struct).generic_types.len > 0 {
-						return_type = left_type
+					left_name := c.table.type_to_str(left_type)
+					right_name := c.table.type_to_str(right_type)
+					if left_name == right_name {
+						c.error('undefined operation `$left_name` $node.op.str() `$right_name`',
+							left_right_pos)
 					} else {
-						left_name := c.table.type_to_str(left_type)
-						right_name := c.table.type_to_str(right_type)
-						if left_name == right_name {
-							c.error('undefined operation `$left_name` $node.op.str() `$right_name`',
-								left_right_pos)
-						} else {
-							c.error('mismatched types `$left_name` and `$right_name`',
-								left_right_pos)
-						}
+						c.error('mismatched types `$left_name` and `$right_name`', left_right_pos)
 					}
 				}
 			} else if right_sym.kind in [.array, .array_fixed, .map, .struct_] {
-				if right_sym.has_method(node.op.str()) {
-					if method := right_sym.find_method(node.op.str()) {
+				if right_sym.has_method_with_generic_parent(node.op.str()) {
+					if method := right_sym.find_method_with_generic_parent(node.op.str()) {
 						return_type = method.return_type
 					} else {
 						return_type = right_type

--- a/vlib/v/checker/tests/generics_undefined_operation.out
+++ b/vlib/v/checker/tests/generics_undefined_operation.out
@@ -1,0 +1,7 @@
+vlib/v/checker/tests/generics_undefined_operation.vv:12:5: error: undefined operation `V2d<int>` * `V2d<int>`
+   10 |     v3 := V2d<int>{4, 6}
+   11 |
+   12 |     if v1 * v2 == v3 {
+      |        ~~~~~~~
+   13 |     }
+   14 | }

--- a/vlib/v/checker/tests/generics_undefined_operation.vv
+++ b/vlib/v/checker/tests/generics_undefined_operation.vv
@@ -1,0 +1,14 @@
+struct V2d<T> {
+mut:
+	x T
+	y T
+}
+
+fn main() {
+	v1 := V2d<int>{2, 2}
+	v2 := V2d<int>{2, 3}
+	v3 := V2d<int>{4, 6}
+
+	if v1 * v2 == v3 {
+	}
+}

